hip/matrix/dense_kernels.hip.cpp

```diff
@@ -303,8 +315,19 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
 template <typename ValueType, typename IndexType>
 void count_nonzero_blocks_per_row(std::shared_ptr<const DefaultExecutor> exec,
                                   const matrix::Dense<ValueType>* source,
-                                  int bs,
-                                  IndexType* result) GKO_NOT_IMPLEMENTED;
+                                  int bs, IndexType* result)
+{
+    const auto num_block_rows = source->get_size()[0] / bs;
+    const auto num_block_cols = source->get_size()[1] / bs;
+    if (num_block_rows > 0) {
+        const auto num_blocks =
+            ceildiv(num_block_rows, default_block_size / config::warp_size);
+        kernel::
+            count_nonzero_blocks_per_row<<<num_blocks, default_block_size>>>(
```

Comment:
See discussion on Slack: It probably makes sense to move to the chevron launch syntax in the future, so I wanted to make a first step in that direction :)

Reply:
I checked the HIP implementation, and the macros already resolve to the Chevron syntax since 2018 inside HIP: https://github.com/ROCm-Developer-Tools/hipamd/blob/250bd582540143f53dc6d61bca963b71cf5a1087/include/hip/hcc_detail/hip_runtime.h
Is that justification enough? :smile: